python/across_api/base/api.py

```diff
@@ -19,3 +23,167 @@
     },
     root_path="/labs/api/v1",
 )
+
+# FastAPI Depends definitions
+
+
+# Depends functions for FastAPI calls.
+async def optional_daterange(
+    begin: Annotated[
+        Optional[datetime],
+        Query(
+            description="Start time of period to be calculated.",
+            title="Begin",
+        ),
+    ] = None,
+    end: Annotated[
+        Optional[datetime],
+        Query(
+            description="Start time of period to be calculated.",
+            title="End",
+        ),
+    ] = None,
```

Comment:
astropy.time.Time can convert from many string formats in addition to datetime objects. Consider broadening types to anything that astropy.time.Time's constructor can accept.

Reply:
This is about interaction with the API endpoints, which always spits out a datetime, so it's not really necessary to support other time formats for this code.

Reply:
I thought that these dependency functions preprocess parameters received from the request. Isn't that so?

Is this in the same category as #2004, something for a future PR?

Reply:
The depends functions are there to create the API endpoint parameters, and drive what ends up in the SwaggerUI interface and the OpenAPI.json specification. OpenAPI supports dates and datetimes in format defined by [RFC 3339, section 5.6](https://tools.ietf.org/html/rfc3339#section-5.6) only. FastAPI converts these into a datetime (naive or not naive depending on whether a timezone is specified in the input). 

The depend function has two main uses:

1. It specifies a standardized parameter name and description for a parameter. By having a function that does this rather than just defining inline, you encourage consistency across the API and code reuse. After all, we're going to end up requesting `begin` and `end` a lot in across many missions. 

2. It converts what FastAPI spits out (in this case a `datetime`) to what we want (an astropy `Time`). 

I believe that our OpenAPI specification should stick with the basic definitions supported, and not use loosely defined strings for simple objects like dates and datetimes. 

Again, this is a design decision, and one we can revisit at a later date. I think it's out of scope here to change it without a detailed look at the pros and cons of how this is done involving the whole team.